src/Home.js

```diff
@@ -0,0 +1,117 @@
+/* eslint-disable jsx-a11y/no-redundant-roles */
+import Header from './components/Header.js';
+import { useState } from 'react';
+import CoinCard from './components/CoinCard.js';
+import styled from 'styled-components';
+import SearchBar from './components/SearchBar.js';
+
+function Home({ coins, currency }) {
+  const [search, setSearch] = useState('');
+  const [searchError, setSearchError] = useState(false);
+
+  const filterdCoins = coins.filter(coin =>
+    coin.name.toLowerCase().includes(search.toLowerCase())
+  );
+
+  const handleChange = e => {
+    setSearch(e.target.value);
+    if (e.target.value !== '') {
+      if (!e.target.value.match(/^[a-zA-Z0-9_ ]+$/)) {
+        setSearchError(true);
+      } else {
+        setSearchError(false);
+      }
+    } else {
+      setSearchError(false);
+    }
+  };
+  return (
+    <>
+      <Header title="Crypto Cloud" />
+      <div className="coins-container">
+        <CoinSearch>
+          <form>
+            <SearchBar handleChange={handleChange} />
+            {searchError ? (
+              <SearchErrorMessage>
+                Please don't use special characters
+              </SearchErrorMessage>
+            ) : (
+              ''
+            )}
+          </form>
+        </CoinSearch>
+        <TableHeading>
+          <p>#</p>
+          <p>Name</p>
+          <StackedHeading>
+            <p>Price</p>
+            <p>% Change 24H</p>
+          </StackedHeading>
+        </TableHeading>
+        <CoinList role="list">
+          {filterdCoins.map(coin => {
+            return (
+              <CoinCard
+                currency={currency}
+                data-testid={'test'}
+                rank={coin.market_cap_rank}
+                key={coin.id}
+                name={coin.name}
+                symbol={coin.symbol}
+                image={coin.image}
+                price={coin.current_price}
+                price_change_percentage_24h={coin.price_change_percentage_24h}
+              />
+            );
+          })}
+          {filterdCoins == [] ? (
```

Comment:
maybe `filterdCoins.length === 0 ?` is better?

Reply:
Thanks for checking mate! I changed the code and added the two improvements.